From 0321ca1e4ab6d176097d90565c7d9502bc19ff48 Mon Sep 17 00:00:00 2001
From: jialelux <jialex.lu@intel.com>
Date: Wed, 24 Aug 2022 02:12:13 +0000
Subject: [PATCH] crypto build error kernel 5.4

---
 scripts/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/Makefile b/scripts/Makefile
index b4b7d8b58..936e70e04 100644
--- a/scripts/Makefile
+++ b/scripts/Makefile
@@ -29,7 +29,7 @@ HOSTCFLAGS_asn1_compiler.o = -I$(srctree)/include
 HOSTCFLAGS_sign-file.o = $(CRYPTO_CFLAGS)
 HOSTLDLIBS_sign-file = $(CRYPTO_LIBS)
 HOSTCFLAGS_extract-cert.o = $(CRYPTO_CFLAGS)
-HOSTLDLIBS_extract-cert = $(CRYPTO_LIBS)
+HOSTLDLIBS_extract-cert = $(CRYPTO_LIBS) -lpthread
 
 always		:= $(hostprogs-y) $(hostprogs-m)
 
-- 
2.25.1

