--- a/configure.ac
+++ b/configure.ac
@@ -224,7 +224,7 @@ AC_ARG_ENABLE(strmax,
 	[case "${enableval}" in
 	@<:@0-9@:>@*) enable_strmax=${enableval} ;;
 	*)      AC_MSG_ERROR([bad value ${enableval} for --enable-strmax]) ;;
-	esac], [enable_strmax="(4UL << 10) /* 4KB */" ])
+	esac], [enable_strmax="(32UL << 10) /* 32KB */" ])
 
 AC_MSG_CHECKING([for --enable-strmax])
 AC_MSG_RESULT([$enable_strmax])
