--- a/src/io/gets_s.c
+++ b/src/io/gets_s.c
@@ -29,6 +29,8 @@
  *------------------------------------------------------------------
  */
 
+#define _GNU_SOURCE
+
 #ifdef FOR_DOXYGEN
 #include "safe_lib.h"
 #else
--- a/src/os/asctime_s.c
+++ b/src/os/asctime_s.c
@@ -29,6 +29,8 @@
  *------------------------------------------------------------------
  */
 
+#define _GNU_SOURCE
+
 #ifdef FOR_DOXYGEN
 #include "safe_lib.h"
 #else
--- a/src/os/ctime_s.c
+++ b/src/os/ctime_s.c
@@ -29,6 +29,8 @@
  *------------------------------------------------------------------
  */
 
+#define _GNU_SOURCE
+
 #ifdef FOR_DOXYGEN
 #include "safe_lib.h"
 #else
--- a/src/os/gmtime_s.c
+++ b/src/os/gmtime_s.c
@@ -29,6 +29,8 @@
  *------------------------------------------------------------------
  */
 
+#define _GNU_SOURCE
+
 #ifdef FOR_DOXYGEN
 #include "safe_lib.h"
 #else
--- a/src/os/localtime_s.c
+++ b/src/os/localtime_s.c
@@ -29,6 +29,8 @@
  *------------------------------------------------------------------
  */
 
+#define _GNU_SOURCE
+
 #ifdef FOR_DOXYGEN
 #include "safe_lib.h"
 #else
